File: FoodOrganizer/app/src/main/java/es/fron99/foodorganize/ViewModels/ActivityLoginVM.java
package es.fron99.foodorganize.ViewModels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class ActivityLoginVM extends ViewModel {

    private MutableLiveData<String> fragmentSelected;
    private MutableLiveData<Boolean> logginOk;

    public ActivityLoginVM(){
        this.fragmentSelected = new MutableLiveData<>();
        this.logginOk = new MutableLiveData<>(false);
    }

    public LiveData<String> getFragmentSelected(){
        if (fragmentSelected == null){
            fragmentSelected = new MutableLiveData<>();
        }
        return fragmentSelected;
    }

    public String getValueFragmentSelected(){
        if (fragmentSelected == null){
            fragmentSelected = new MutableLiveData<>();
        }
        return fragmentSelected.getValue();
    }

    public void changeFragmentSelected(String newValue){
        if (fragmentSelected == null){
            fragmentSelected = new MutableLiveData<>();
        }
        fragmentSelected.setValue(newValue);
    }

    public LiveData<Boolean> getLogginOk(){
        if (logginOk == null){
            logginOk = new MutableLiveData<>();
        }
        return logginOk;
    }

    public Boolean getValueLogginOk(){
        if (logginOk == null){
            logginOk = new MutableLiveData<>();
        }
        return logginOk.getValue();
    }

    public void changeLogginOk(Boolean newValue){
        if (logginOk == null){
            logginOk = new MutableLiveData<>();
        }
        logginOk.setValue(newValue);
    }

}

File: FoodOrganizer/app/src/main/java/es/fron99/Foodorganize/ViewModels/ActivityLoginVM.kt
package es.fron99.Foodorganize.ViewModels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

@Suppress("unused")
class ActivityLoginVM : ViewModel() {

    /************************************************FragmentSelected************************************************/

    private var fragmentSelected: MutableLiveData<String> = MutableLiveData()

    fun getFragmentSelected(): LiveData<String> {
        return this.fragmentSelected
    }

    fun valueFragmentSelected(): String? {
        return fragmentSelected.value
    }

    fun changeFragmentSelected(newValue: String) {
        fragmentSelected.value = newValue
    }

    /************************************************LogginOk************************************************/

    private var logginOk: MutableLiveData<Boolean> = MutableLiveData(false)

    fun getLogginOk(): LiveData<Boolean> {
        return this.logginOk
    }

    fun getvalueLogginOk(): Boolean? {
        return logginOk.value
    }

    fun changeLogginOk(newValue: Boolean) {
        logginOk.value = newValue
    }

}